direnv/config.nu

```diff
@@ -0,0 +1,16 @@
+## To enable direnv in nushell we must run code in the context of the current shell - i.e it cannot be within a block and it needs to run as a "code" string as per https://github.com/nushell/nushell/pull/5982 (so you must run nushell 0.66 or later). That way it works as if you'd typed in and run the code directly in your shell.
+## Direnv knows what to do otherwise and will export the env to load (or unload) based on what is in your current environment so this is all that is needed with some checks for empty strings, defaulting then to an empty table so that load-env is always happy.
+
+let-env config = {
+  hooks: {
+    env_change: {
+      PWD: [{
+        code: "
+          let direnv = (direnv export json | from json)
+          let direnv = if ($direnv | length) == 1 { $direnv } else { {} }
+          $direnv | load-env
```

Comment:
how does load-env work? Does it unset env vars if their value is null?

Reply:
if an env var is set to `$nothing` or `null` it appears to be unset.
```
> {NAME: "Test", AGE: 1} | load-env
> $env.NAME
Test
> $env.AGE
1
> {NAME: $nothing, age: null} | load-env
> $env.NAME <-- empty return
> $env.AGE <-- empty return
```

Reply:
@zimbatm It will set the value to null/nothing. You can query it but the value will be nothing, so it's not unsetting. To unset an environment variable, use `hide`. (I'm planning to make a `hide-env` for unsetting only environment variables specifically which will be more powerful than the current `hide`.)

Reply:
How does nushell treat non-string env vars? I assume the null env vars get cast to empty strings and numbers to their equivalent string representation?

If nushell only allowed strings as valid values, it could assume that null means hide.